core/capabilities/ccip/configs/solana/chain_writer_test.go

```diff
@@ -26,11 +30,18 @@ func TestChainWriterConfigRaw(t *testing.T) {
 
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			_, err := GetSolanaChainWriterConfig("4Nn9dsYBcSTzRbK9hg9kzCUdrCSkMZq1UR6Vw1Tkaf6H", tt.fromAddress, 0)
+			config, err := GetSolanaChainWriterConfig("4Nn9dsYBcSTzRbK9hg9kzCUdrCSkMZq1UR6Vw1Tkaf6H", tt.fromAddress, 0)
 			if tt.expectedError != "" {
 				assert.EqualError(t, err, tt.expectedError)
 			} else {
-				assert.NoError(t, err)
+				require.NoError(t, err)
+
+				raw, err := json.Marshal(config)
+				require.NoError(t, err)
+				var result chainwriter.ChainWriterConfig
+				err = json.Unmarshal(raw, &result)
+				require.NoError(t, err)
+				require.EqualValues(t, config, result)
```

Comment:
This is great, we have a test now to validate if serialization/deserialization is not a lossy process!